orcid-message-listener/src/main/java/org/orcid/listener/listeners/updated/UpdatedOrcidWorker.java

```diff
@@ -37,31 +40,69 @@
 
     Logger LOG = LoggerFactory.getLogger(UpdatedOrcidWorker.class);
 
+    @Value("${org.orcid.persistence.messaging.dump_indexing.enabled}")
+    private boolean dumpIndexingEnabled;
+    
+    @Value("${org.orcid.persistence.messaging.solr_indexing.enabled}")
+    private boolean solrIndexingEnabled;
+    
     @Resource
     private Orcid12APIClient orcid12ApiClient;
     @Resource
+    private Orcid20APIClient orcid20ApiClient;
+    @Resource
     private SolrIndexUpdater solrIndexUpdater;
     @Resource
     private S3Updater s3Updater;
 
     /**
-     * Fetches ORCID profile, compares last modified with SOLR if after SOLR
-     * last modified, updates SOLR index TODO: update S3
-     * 
+     * Populates the Amazon S3 buckets and updates solr index
      */
     public void onRemoval(RemovalNotification<String, LastModifiedMessage> removal) {
         if (removal.wasEvicted()) {
             LastModifiedMessage m = removal.getValue();
             LOG.info("Removing " + removal.getKey() + " from UpdatedOrcidCacheQueue");
             LOG.info("Processing " + m.getLastUpdated());
-            OrcidProfile profile = orcid12ApiClient.fetchPublicProfile(m.getOrcid());
-            Date lastModifiedFromprofile = profile.getOrcidHistory().getLastModifiedDate().getValue().toGregorianCalendar().getTime();
-            Date lastModifiedFromSolr = solrIndexUpdater.retrieveLastModified(m.getOrcid());
-            // note this is slightly different from existing behaviour
-            if (lastModifiedFromprofile.after(lastModifiedFromSolr))
-                solrIndexUpdater.updateSolrIndex(profile);
-            // now do the same for S3...
-            s3Updater.updateS3(profile);
+            String orcid = m.getOrcid();
+            
+            // Phase #1: update S3 
```

Comment:
Note: when we start to do SOLR, fetch the profile here so we don't have to fetch it twice.